Gq generator

wonkaba opened the pull request:
Most code works fine. Weird bug in a date creation method that I want you to put a fresh set of eyes on. Pull this so you can look at it but don't merge it into master yet. Call me in the AM. We should get together and work.

wonkaba committed:
OK...
Added QR generator
Qr generator returns qr image as UIImage back to professor create/view page via delegate and protocol
Added logic to take a date strings from the qr code and convert to NSDate objects
Added logic to compare Begin date/time and End date/time of office hours to the current date/time which returns a boolean value
committing this before I start to implement actual QR reading of date/time and implementing conversions and checks

wonkaba committed:
implemented code to check date but crashing due to optional unwrapping bug in StudentViewController
Commented out bad code and inserts fine.
Needs de-bug

gmoalvarez committed:
added playground with working date

gmoalvarez commented:
I just uploaded a playground with a working date from string converter:

```
let currentDate = NSDate()
let dateInString = "2015-11-21 13:10:15"  //sample date string
let dateFmt = NSDateFormatter()
dateFmt.timeZone = NSTimeZone.defaultTimeZone()
dateFmt.dateFormat = "yyyy-MM-dd HH:mm:ss"
let returnDate = dateFmt.dateFromString(dateInString)
```

gmoalvarez committed:
small changes to date

wonkaba committed:
makeDate()  and timeCheck working with 24 hour mode
making the new queue segues to qrCodeGenerator
currently the queueID is hardcoded to send to qrCodeGenerator
needs to be re-written in createQueue() method of ProfessorCreateOrViewController
qrCodeGenerator currently hard codes the time of office hours to be Begin: 2015-11-21 12:01  End: 2015-12-21 03:00 so that it won't fail unless we want it to.

gmoalvarez committed:
queue is created and queue object Id is correctly sent to the QR Code generator to create the qr code. There is a problem converting the QR code to data to upload to Parse.
UIImageJPEGRepresentation fails to convert

gmoalvarez committed:
Professor View Queue now shows latest queue

wonkaba committed:
added pics

wonkaba committed:
images
